### PR TITLE
mgr/dashboard: carbonize the wizard component

### PR DESCRIPTION
**Pull request by nizamial09**
Fixes: https://tracker.ceph.com/issues/68103

Screenshot from 2024-09-17 15-02-58
image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by Pegonzal — COMMENTED** [review]
From the testing side, looks mostly fine but the `Create OSDs` page still uses bootstrap, so it doesn't fully have the carbonized feel

Video recording:

[screen-capture (25).webm](https://github.com/user-attachments/assets/257639ea-9150-4bcd-8ec9-67c7792f3a93)

**Comment by nizamial09** [comment]
> but the `Create OSDs` page still uses bootstrap, so it doesn't fully have the carbonized feel

@Pegonzal that is expected, we actually need to carbonize the OSDs form component, which I am not doing in this PR. Let;s deal  with it separately. to trim down the scope, i only did the generic wizard component.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts [hunk 1]
```diff
@@ -125,6 +125,9 @@ export class TableComponent implements AfterViewInit, OnInit, OnChanges, OnDestr
   @Input()
   hasDetails = false;
 
+  @Input()
+  showMenu = true;
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts:129]:
```suggestion
  showToolbar = true;
```

Would that be better? I found it kinda confusing.
Referring to: https://carbondesignsystem.com/components/data-table/usage/#anatomy

**Review by Pegonzal — APPROVED** [review]
Thanks @nizamial09  LGTM, I left a couple NITS
Also on the UI side I think some encapsulation in the components is missing, some kind of grid or table, like we used to have and like it can be seen in the guidelines here: 
https://carbondesignsystem.com/components/data-table/usage/#anatomy

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts [hunk 2]
```diff
@@ -55,6 +72,7 @@ export class CreateClusterComponent implements OnInit, OnDestroy {
   selectedOption = {};
   simpleDeployment = true;
   stepsToSkip: { [steps: string]: boolean } = {};
+  steps: Observable<WizardStepModel[]>;
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts:75]:
```suggestion
```
Is this being used?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts [hunk 3]
```diff
@@ -16,20 +16,33 @@ import _ from 'lodash';
...
+      label: 'Create Realm & Zonegroup'
+    },
+    {
+      label: 'Create Zone'
+    },
+    {
+      label: 'Select Cluster'
+    }
+  ];
+  // stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Select Cluster'];
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts]: removed

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts]: Thanks @Pegonzal, I checked that doc and in the context of carbon datatable, those are called inline actions. the toolbar is something that holds the search bar and filters along with the primary action.
https://carbondesignsystem.com/components/data-table/usage/#inline-actions

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts (hunk 2 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts]: removed, thanks

**Comment by nizamial09** [comment]
> in the components is missing, some kind of grid or table, like we used to have and like it can be seen in the guidelines here:

agreed, i tried to do this but i am a bit lost on the carbon implementation. I am gonna need the help of @ivoalmeida to actually fix the issues related to it.

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Pull request merged**